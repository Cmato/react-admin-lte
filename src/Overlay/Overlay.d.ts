import * as React from 'react';

export interface OverlayProps {
  dark?: boolean;
}

declare const Overlay: React.ComponentType<OverlayProps>;

export default Overlay;
