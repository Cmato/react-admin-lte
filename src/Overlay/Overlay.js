import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Overlay = ({
  children,
  className: classNameProp,
  dark,
  disableWrapper,
  isOverlay: _,
  ...other
}) => {
  if (disableWrapper) {
    return (
      <div
        className={classNames({ overlay: true, dark }, classNameProp)}
        {...other}
      >
        {children}
      </div>
    );
  } else {
    return (
      <div
        className={classNames({ 'overlay-wrapper': true }, classNameProp)}
        {...other}
      >
        <div className={classNames({ overlay: true, dark })}>
          {children}
        </div>
      </div>
    );
  }
};

Overlay.propTypes = {
  /**
   * The contents of the component.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * If `true`, the overlay will have a dark background.
   */
  dark: PropTypes.bool,
  /**
   * @ignore
   */
  disableWrapper: PropTypes.bool,
  /**
   * @ignore
   */
  isOverlay: PropTypes.bool
};

Overlay.defaultProps = {
  dark: false,
  disableWrapper: false,
  isOverlay: true
};

export default Overlay;
