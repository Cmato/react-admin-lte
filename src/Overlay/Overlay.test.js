import React from 'react';
import { shallow } from 'enzyme';
import Overlay from './Overlay';

const spinner = <div>spinner</div>;

it('renders default', () => {
  expect(shallow(<Overlay>{spinner}</Overlay>)).toMatchSnapshot();
});

it('renders disabled wrapper', () => {
  expect(shallow(<Overlay disableWrapper={true}>{spinner}</Overlay>))
    .toMatchSnapshot();
});

it('renders dark with disabled wrapper', () => {
  expect(shallow((
    <Overlay
      dark={true}
      disableWrapper={true}
    >
      {spinner}
    </Overlay>
  ))).toMatchSnapshot();
});

it('renders dark', () => {
  expect(shallow((
    <Overlay
      dark={true}
    >
      {spinner}
    </Overlay>
  ))).toMatchSnapshot();
});
