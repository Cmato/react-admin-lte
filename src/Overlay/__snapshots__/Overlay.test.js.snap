// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark 1`] = `
<div
  className="overlay-wrapper"
>
  <div
    className="overlay dark"
  >
    <div>
      spinner
    </div>
  </div>
</div>
`;

exports[`renders dark with disabled wrapper 1`] = `
<div
  className="overlay dark"
>
  <div>
    spinner
  </div>
</div>
`;

exports[`renders default 1`] = `
<div
  className="overlay-wrapper"
>
  <div
    className="overlay"
  >
    <div>
      spinner
    </div>
  </div>
</div>
`;

exports[`renders disabled wrapper 1`] = `
<div
  className="overlay"
>
  <div>
    spinner
  </div>
</div>
`;
